### PR TITLE
FIX: error message when redis is oom or busy or loading

### PR DESCRIPTION
**Pull request by liguangbo**
image

### CODE REVIEW

File: src/nc_response.c [hunk 1]
```diff
@@ -193,12 +193,12 @@ rsp_filter(struct context *ctx, struct conn *conn, struct msg *msg)
      * If auto_eject_host is enabled, this will also update the failure_count
      * and eject the server if it exceeds the failure_limit
      */
-    if (msg->failure(msg)) {
+    if (msg->failure(msg) != 0) {
```
Comment (git-hulk) [inline src/nc_response.c]: clean-tidy:
```
if ((err = msg->failure(msg)) != 0) {
    ...
    conn->err = err;
}
```

File: src/proto/nc_redis.c [hunk 2]
```diff
@@ -2385,13 +2385,13 @@ redis_failure(struct msg *r)
     case MSG_RSP_REDIS_ERROR_OOM:
     case MSG_RSP_REDIS_ERROR_BUSY:
     case MSG_RSP_REDIS_ERROR_LOADING:
-        return true;
+        return -(r->type);
```
Comment (git-hulk) [inline src/proto/nc_redis.c]: => -1 * r->type

File: src/proto/nc_redis.c [hunk 3]
```diff
@@ -3064,3 +3064,21 @@ redis_swallow_msg(struct conn *conn, struct msg *pmsg, struct msg *msg)
                  conn_server->name.data, message);
     }
 }
+
+char*
+redis_failure_msg(err_t err)
+{
+    switch (-err) {
```
Comment (git-hulk) [inline src/proto/nc_redis.c]: => -1 * err

File: src/nc_response.c (hunk 1 above)
Comment (liguangbo) [reply src/nc_response.c]: fix

File: src/proto/nc_redis.c (hunk 2 above)
Comment (liguangbo) [reply src/proto/nc_redis.c]: fix

File: src/proto/nc_redis.c (hunk 3 above)
Comment (liguangbo) [reply src/proto/nc_redis.c]: fix

**Review by git-hulk — APPROVED** [review]

**Pull request merged**